incubator/command-log/src/main/java/io/aklivity/zilla/runtime/command/log/internal/LoggableStream.java

```diff
@@ -387,6 +390,21 @@ private void onData(
             sequence - acknowledge + reserved, maximum, format("DATA [0x%016x] [%d] [%d] [%x] [0x%016x]",
                     budgetId, length, reserved, flags, authorization));
 
+        if (withPayload)
+        {
+            final OctetsFW payload = data.payload();
+            if (payload != null)
+            {
+                StringBuilder hexData = new StringBuilder();
+                for (int i = 0; i < data.length(); i++)
+                {
+                    hexData.append(String.format("%02x", payload.buffer().getByte(payload.offset() + i)));
+                    hexData.append(i < data.length() - 1 ? ':' : "");
+                }
+                out.printf(verboseFormat, index, offset, timestamp, hexData);
```

Comment:
Please take a look at using `BitUtil.toHex(...)` with a shared temporary byte array instead of string builder append.

Reply:
Use `payload.buffer.getBytes(...)` to populate the `byte[]`.